native/src/snappyclient/cpp/impl/SSLParameters.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * SSLParameters.cpp
+ *
+ *  Created on: 13-Dec-2019
+ *      Author: pbisen
+ */
+
+//#include "thrift/snappydata_struct_SnappyException.h"
+#include "SSLParameters.h"
+using namespace io::snappydata;
+using namespace io::snappydata::client::impl;
+using namespace io::snappydata::thrift;
+
+void SSLParameters::operator()(const std::string& str) {
+  size_t spos;
+  if ((spos = str.find('=')) != std::string::npos) {
+    std::string propertyValue = str.substr(spos + 1);
+    std::string propertyName = str.substr(0, spos);
+    setSSLProperty(propertyName, propertyValue);
+    return;
+  }
+}
+
+void SSLParameters::setSSLProperty(std::string &propertyName,
+    std::string& value) {
+  auto itr = sslProperties.find(propertyName);
+  if (itr != sslProperties.end()) {
+    auto mapItr = sslPropValMap.find(propertyName);
+    if (mapItr == sslPropValMap.end()) {
+      sslPropValMap.insert(
+          std::pair<std::string, std::string>(propertyName, value));
+    }
+  }else{
+    throw std::invalid_argument(":Unknown SSL Property:" + propertyName );
+  }
+}
+void SSLParameters::getSSLPropertyValue(std::string &propertyName,
+    std::string& value) {
+  auto itr = sslProperties.find(propertyName);
+  if (itr != sslProperties.end()) {
+    auto itr = sslPropValMap.find(propertyName);
+    if (itr != sslPropValMap.end()) {
+      value = itr->second;
+    }
```

Comment:
Return value instead of by reference and empty string for the else case above.

Reply:
Resolved